src/test/java/uk/gov/ons/ctp/common/event/EventPublisherTest.java

```diff
@@ -266,28 +226,41 @@ public void shouldSentAddressNotValid() throws Exception {
     verify(sender).sendEvent(eq(routingKey), eventCapture.capture());
     AddressNotValidEvent event = eventCapture.getValue();
 
-    assertEquals(event.getEvent().getTransactionId(), transactionId);
-    assertThat(UUID.fromString(event.getEvent().getTransactionId()), instanceOf(UUID.class));
-    assertEquals(EventPublisher.EventType.ADDRESS_NOT_VALID, event.getEvent().getType());
-    assertEquals(EventPublisher.Source.CONTACT_CENTRE_API, event.getEvent().getSource());
-    assertEquals(EventPublisher.Channel.CC, event.getEvent().getChannel());
-    assertThat(event.getEvent().getDateTime(), instanceOf(Date.class));
+    assertHeader(
+        event, transactionId, EventType.ADDRESS_NOT_VALID, Source.CONTACT_CENTRE_API, Channel.CC);
+    assertEquals(payload, event.getPayload().getInvalidAddress());
+  }
+
+  private void assertSendCase(EventType type) {
+    CollectionCase payload = EventFixture.createCollectionCase();
+
+    ArgumentCaptor<CaseEvent> eventCapture = ArgumentCaptor.forClass(CaseEvent.class);
 
-    AddressNotValid eventPayload = event.getPayload().getInvalidAddress();
+    String transactionId =
+        eventPublisher.sendEvent(type, Source.CONTACT_CENTRE_API, Channel.CC, payload);
+
+    RoutingKey routingKey = RoutingKey.forType(type);
+    verify(sender).sendEvent(eq(routingKey), eventCapture.capture());
+    CaseEvent event = eventCapture.getValue();
 
-    assertEquals(collectionCase.getId(), eventPayload.getCollectionCase().getId());
-    assertEquals("buy weetabix", eventPayload.getNotes());
-    assertEquals("DERELICT", eventPayload.getReason());
+    assertHeader(event, transactionId, type, Source.CONTACT_CENTRE_API, Channel.CC);
+    assertEquals(payload, event.getPayload().getCollectionCase());
+  }
+
+  @Test
+  public void shouldSendCaseCreated() {
+    assertSendCase(EventType.CASE_CREATED);
   }
 
-  /** Test event message with FeedbackResponse payload */
   @Test
-  public void sendEventFeedbackPayload() throws Exception {
+  public void shouldSendCaseUpdated() {
+    assertSendCase(EventType.CASE_UPDATED);
+  }
 
-    Feedback feedbackResponse = new Feedback();
-    feedbackResponse.setPageUrl("url-x");
-    feedbackResponse.setPageTitle("randomPage");
-    feedbackResponse.setFeedbackText("Bla bla bla");
+  /** Test event message with FeedbackResponse payload */
+  @Test
+  public void sendEventFeedbackPayload() {
+    Feedback feedbackResponse = EventFixture.createFeedback();
```

Comment:
I can't see why breaking out the code into a separate class is better.  In order to read the code you've got to jump back and forward between classes.
I guess that we'll disagree on this point. Perhaps opening up into a wider discussion

Reply:
First, sorry that this PR had more code changes than you were expecting!

So we discussed this , and I'll just try to summarise my approach:

- there is value in having smaller test methods to help focus on what is being achieved
- in this instance the tests are all about ensuring certain event types are sent, that the headers are populated correctly and the payload goes across. None of this relies on the specific data going into the model objects, which are all annotated with lombok Data and can easily be compared with assertEquals.
- there is value in having Fixture helper classes to build model objects with data , most or all of which is not interesting to the thing being tested
- it is good to separate concerns of classes: in this case we have a Fixture class focused on model object building for testing, and the actual test class focused testing the sending
- I appreciate there is less value (at present) for the Fixture class because of minimum reuse right now, but that in all likelihood will change (see below)
- there is value in refactoring things "as you go" based on the fact that in many times you seldom get the luxury to have specific separate refactoring tasks. I now understand that CR-000 type PR's are possible, but it is still something that is less likely to be done because of the separate nature.
- as we've talked about today, this project will most likely expand quite a bit with the work needed by CR-997, so there will be more unit tests, and the fixtures will be reused more
- at present there is very low code coverage on this project, so there is a likelihood that future work will need more tests, and then make use of the fixture class pre-baked model objects.

I understand that the meat of this PR is in the CollectionCase class which is just a few lines, but I figured it needed some additional unit tests, and I took a decision to try to tidy things.